devito/types/basic.py

```diff
@@ -938,6 +942,10 @@ def _C_typename(self):
     def _C_ctype(self):
         return self.dtype
 
+    @property
+    def function(self):
```

Comment:
if you add this here, there's a chance we can drop something from subclasses? or maybe not?

Reply:
No subclass implements it